akka-http/src/main/scala/akka/http/scaladsl/server/HttpApp.scala

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (C) 2009-2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.http.scaladsl.server
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.stream.ActorMaterializer
+import akka.http.scaladsl.settings.ServerSettings
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.{ Await, ExecutionContext, Future }
+import scala.io.StdIn
+import scala.util.{ Failure, Success, Try }
+
+/**
+ * Bootstrap trait. It helps booting up an akka-http by only defining the desired routes.
+ * It offers additional hooks to modify the default behavior
+ */
+trait HttpApp {
+
+  /**
+   * Start a server on the specified host and port, using provided settings and [[ActorSystem]] if present
+   */
+  def startServer(host: String, port: Int, settings: ServerSettings, system: ActorSystem = null): Unit = {
+    // Having a null is suboptimal, but [[Option]] might be too verbose?.
+    // Having null or [[Option]] is also useful to shutdown the system if we created it
+    implicit val theSystem = if (system == null) ActorSystem(getClass.getSimpleName) else system
+    implicit val materializer = ActorMaterializer()
+    implicit val executionContext = system.dispatcher
```

Comment:
One more reason to stick to Option :) https://github.com/akka/akka-http/pull/542#discussion_r94954359